core/src/main/java/io/opencensus/stats/View.java

```diff
@@ -42,20 +48,43 @@
   public abstract Measure getMeasure();
 
   /**
+   * The {@link Aggregation}s associated with this {@link View}.
+   */
+  public abstract List<Aggregation> getAggregations();
+
+  /**
    * Columns (a.k.a Tag Keys) to match with the associated {@link Measure}.
-   * If no dimensions are specified, then all stats are recorded. Columns must be unique.
    *
-   * <p>Note: The returned list is unmodifiable, attempts to update it will throw an
-   * UnsupportedOperationException.
+   * <p>{@link Measure} will be recorded in a "greedy" way. That is, every view aggregates every
+   * measure. This is similar to doing a GROUPBY on view’s columns. Columns must be unique.
    */
   public abstract List<TagKey> getColumns();
 
   /**
-   * Applies the given match function to the underlying data type.
+   * Returns the time {@link Window} for this {@code View}.
+   *
+   * @return the time {@link Window}.
    */
-  public abstract <T> T match(
-      Function<? super DistributionView, T> p0,
-      Function<? super IntervalView, T> p1);
+  public abstract Window getWindow();
+
+  /**
+   * Constructs a new {@link View}.
+   */
+  public static View create(
+      Name name,
+      String description,
+      Measure measure,
+      List<Aggregation> aggregations,
```

Comment:
I think it would be confusing to require users to rely on the order in a LinkedHashSet when matching TagKeys with a list of TagValues or matching Aggregations with a list of AggregationDatas.  Why don't we just check for duplicates in the input TagKey and Aggregation lists?  We only need to check once, because the data structures are immutable.  Lists make it clear that the order matters.

Another option is to allow users to provide unordered Sets of TagKeys and Aggregations in Views.  Then we could return data structures that pair the inputs and outputs, i.e., we could return TagValues as a `List<Tag>` or return AggregationDatas as `Map<Aggregation, AggregationData>`.  Users wouldn't need to match the objects in the View with objects in the ViewData.